calendar-bundle/src/Resources/contao/classes/Events.php

```diff
@@ -423,73 +419,25 @@ protected function addEvent($objEvents, $intStart, $intEnd, $intBegin, $intLimit
 	 * @param boolean             $blnAbsolute
 	 *
 	 * @return string
+	 *
+	 * @deprecated Deprecated since Contao 4.10. Use the Symfony router instead.
 	 */
 	public static function generateEventUrl($objEvent, $blnAbsolute=false)
 	{
-		$strCacheKey = 'id_' . $objEvent->id . ($blnAbsolute ? '_absolute' : '');
+		@trigger_error(__METHOD__.' is deprecated since Contao 4.10. Use the Symfony router instead.', E_USER_DEPRECATED);
```

Comment:
Use `trigger_deprecation` here (see #1946).

Reply:
Fixed in 5db43fe2d7f72fb92f3e92804f74419d6fe17ba8

Reply:
fixed in 7cbf8f03f78304f8e46128b5ab57b60bcd36a45e